### PR TITLE
DIV-2808

### PR DESCRIPTION
**Pull request by niyioyelade**
Upgrade Spring Boot Version - Upgraded to SpringBoot 2 successfully

# Description
https://tools.hmcts.net/jira/browse/DIV-2808

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: dependency-check-suppressions.xml [hunk 1]
```diff
@@ -8,4 +8,9 @@
         <gav regex="true">^org\.apache\.tomcat:tomcat-annotations-api:.*$</gav>
         <cpe>cpe:/a:apache:tomcat</cpe>
     </suppress>
+    <suppress>
+        <gav regex="true">.*</gav>
```
Comment (tchow8) [inline dependency-check-suppressions.xml]: Can you confirm these CVEs are safe to suppress?

Comment (niyioyelade) [reply dependency-check-suppressions.xml]: I see we suppress them in other projects as well. I believe it's safe to do so.

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -39,27 +49,19 @@ management:
       enabled: true
     diskspace:
       enabled: true
-  security:
-    enabled: false
+  endpoint:
+    health:
+      enabled: true
+      cache:
+        time-to-live: 1000
```
Comment (mthava) [inline src/main/resources/application.yml]: can you change this to 5000

File: src/test/java/uk/gov/hmcts/reform/divorce/feepayment/controller/FeePaymentServiceControllerTest.java [hunk 3]
```diff
@@ -5,14 +5,12 @@
...
+import org.mockito.junit.MockitoJUnitRunner;
 import uk.gov.hmcts.reform.divorce.feepayment.model.Fee;
+import uk.gov.hmcts.reform.divorce.feepayment.service.FeePaymentService;
 
-import static org.mockito.Matchers.any;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
```
Comment (mthava) [inline src/test/java/uk/gov/hmcts/reform/divorce/feepayment/controller/FeePaymentServiceControllerTest.java]: Can you please expand the imports. PMD should have complained about this. You can change it in IntelliJ.

File: src/main/java/uk/gov/hmcts/reform/divorce/feepayment/service/impl/FeePaymentServiceImpl.java [hunk 4]
```diff
@@ -56,4 +56,7 @@ private Fee extractValue(ObjectNode objectNode) {
     }
 
 
+    public void setFeeApiUrl(String feeApiUrl) {
```
Comment (mthava) [inline src/main/java/uk/gov/hmcts/reform/divorce/feepayment/service/impl/FeePaymentServiceImpl.java]: Remove this. And use reflection util to inject the api url. Some thing like ReflectionTestUtils.setField(classUnderTest, "jurisdictionId", JURISDICTION_ID);

File: src/test/java/uk/gov/hmcts/reform/divorce/feepayment/service/FeePaymentServiceTest.java [hunk 5]
```diff
@@ -1,25 +1,37 @@
...
+    private RestTemplate restTemplate;
+    private String feeApiUrl = "http://feeApiUrl";
 
+    @InjectMocks
+    private FeePaymentServiceImpl feePaymentService;
     @Before
     public void setup() {
         assertNotNull(feePaymentService);
+        feePaymentService.setFeeApiUrl(feeApiUrl);
+        Mockito.when(restTemplate.getForObject(Mockito.any(URI.class), Mockito.any(Class.class))).thenReturn(new ObjectNode[]{});
```
Comment (mthava) [inline src/test/java/uk/gov/hmcts/reform/divorce/feepayment/service/FeePaymentServiceTest.java]: User the real values when mocking. Do not use any() unless it is absolutely necessary.

File: src/test/resources/application.yml [hunk 6]
```diff
@@ -60,15 +60,13 @@ management:
...
   health:
     enabled: true
-    sensitive: false
-    id: health
-    time-to-live: 1000
+    cache:
+      time-to-live: 1000
   metrics:
     enabled: true
-    sensitive: false
```
Comment (mthava) [inline src/test/resources/application.yml]: I dont think we need a seperate application.yml for test. Could have used the sources one.

File: src/main/java/uk/gov/hmcts/reform/divorce/feepayment/service/impl/FeePaymentServiceImpl.java (hunk 4 above)
Comment (niyioyelade) [reply src/main/java/uk/gov/hmcts/reform/divorce/feepayment/service/impl/FeePaymentServiceImpl.java]: Done

File: src/test/resources/application.yml (hunk 6 above)
Comment (niyioyelade) [reply src/test/resources/application.yml]: Will leave this in for now as it gives us more configuration flexibility around writing test cases.

File: src/test/java/uk/gov/hmcts/reform/divorce/feepayment/controller/FeePaymentServiceControllerTest.java (hunk 3 above)
Comment (niyioyelade) [reply src/test/java/uk/gov/hmcts/reform/divorce/feepayment/controller/FeePaymentServiceControllerTest.java]: Done

File: src/main/resources/application.yml (hunk 2 above)
Comment (niyioyelade) [reply src/main/resources/application.yml]: Done

**Pull request merged**